### PR TITLE
Added error handling and Input Validation for Robust Web Scraping

### PR DESCRIPTION
**Pull request by tamannaaaaa**
## Description

Currently, the web scraper does not include error handling for potential network issues or server errors during the HTTP request to the BBC News website. To make the scraper more resilient and prevent crashes, we should implement appropriate error handling using try-except blocks. Additionally, it would be beneficial to add input validation to the BBCNews class to ensure proper input and avoid unexpected behavior in case of future parameter additions. This enhancement will improve the scraper's reliability and maintainability.

## Checklist

> Before submitting this pull request, kindly verify that the ensuing checkpoints have been reached.

- [x] Have you adhered to the repository's defined coding convention rules? Yes
- [ ] Have you updated the 'documentation.md' file with the method/function documentation? No
- [x] Have you sent a message along with the result or response?  Yes
- [ ] Have you used the try-catch technique? Yes
- [x] Has the method/class been added to the documentation (md file)? No

I certify that I have carried out the relevant checks and provided the requisite screenshot for validation by submitting this pull request.
I appreciate your contribution.

### CODE REVIEW

File: first-contribution
Comment (nikhil25803) [inline first-contribution:1]: I think you have also pushed any sub-module of the first-contribution repository. Check once

**Review by nikhil25803 — CHANGES_REQUESTED** [review]

**Review by tamannaaaaa — COMMENTED** [review]
I have made my changes

**Review by nikhil25803 — CHANGES_REQUESTED** [review]
image

It is still there and will be added once the PR is merged.
I will recommend you to close this PR and make a new one with proper changes.

**Comment by tamannaaaaa** [comment]
Ok I will follow your instructions.

**Comment by tamannaaaaa** [comment]
I don't know why but I'm unable to delete that submodule

**Comment by nikhil25803** [comment]
I think you should delete the folder named `first-contribution`.

**Comment by tamannaaaaa** [comment]
I have done it. Please check and let me know if any deletion or update I need to do

**Comment by tamannaaaaa** [comment]
It is the new pull request "New branches"

**Pull request closed without merging**